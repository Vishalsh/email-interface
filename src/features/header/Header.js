import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/button/Button';
import classes from './Header.module.scss';

const Header = ({ onClickHamburger }) => {
  return (
    <header className={classes.header}>
      <div className="row">
        <div className="col-xs-6">
          <div className="row middle-xs">
            <Button type="primary" onClick={onClickHamburger}>
              <i className="icon icon-hamburger"/>
            </Button>
            <input className={classes.search} placeholder="Search for something..."/>
          </div>
        </div>

        <div className="col-xs-6">
          <div className="row end-xs middle-xs">
            <i className="icon icon-email-solid"/>
            <div>
              <Button onClick={() => alert('logout')}>
                <i className="icon icon-logout"/>
              </Button>
              <span>Log out</span>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  onClickHamburger: PropTypes.func.isRequired,
};

export default Header;
