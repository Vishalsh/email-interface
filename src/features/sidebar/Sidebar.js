import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import InboxEmailsCount from 'features/mailbox/inboxEmailsCount/InboxEmailsCount';
import UserAvatar from 'components/userAvatar/UserAvatar';
import SidebarLink from './sidebarLink/SidebarLink';
import routes from 'constants/routes';
import classes from './Sidebar.module.scss';

const links = [{
  name: 'logo',
  icon: 'logo',
  routeTo: '',
  disabled: true
}, {
  name: 'Dashboard',
  icon: 'dashboard',
  routeTo: routes.DASHBOARD
}, {
  name: 'Layouts',
  icon: 'layouts',
  routeTo: routes.LAYOUTS
}, {
  name: 'Graphs',
  icon: 'graphs',
  routeTo: routes.GRAPHS
}, {
  name: 'Mailbox',
  icon: 'mailbox',
  routeTo: routes.MAILBOX,
  links: [{
    name: 'Inbox',
    routeTo: routes.INBOX
  }, {
    name: 'Email view',
    routeTo: routes.EMAIL_VIEW,
    disabled: true
  }, {
    name: 'Compose Email',
    routeTo: routes.COMPOSE_EMAIL
  }, {
    name: 'Email templates',
    routeTo: routes.EMAIL_TEMPLATES,
    disabled: true
  }]
}, {
  name: 'Metrics',
  icon: 'metrics',
  routeTo: routes.METRICS
}, {
  name: 'Widgets',
  icon: 'widgets',
  routeTo: routes.WIDGETS
}, {
  name: 'Forms',
  icon: 'forms',
  routeTo: routes.FORMS
}, {
  name: 'App Views',
  icon: 'app-views',
  routeTo: routes.APP_VIEWS
}];

export const Sidebar = (props) => {
  const { isOpen, user } = props;
  const { name, role, avatar } = user;

  return (
    <div className={`${classes.sidebar} ${isOpen ? classes.sidebarOpen : ''}`}>
      <nav className={classes.nav}>
        {
          isOpen ?
            <div className={`${classes.avatar} ${isOpen ? classes.avatarOpen : ''}`}>
              <UserAvatar name={name} role={role} avatar={avatar}/>
            </div>
            :
            <SidebarLink link={links[0]}/>
        }

        <SidebarLink link={links[1]}/>
        <SidebarLink link={links[2]}/>
        <SidebarLink link={links[3]}/>
        <SidebarLink link={links[4]}
                     isSidebarOpen={isOpen}
                     rightNode={<InboxEmailsCount showReadMailsCount />}/>
        <SidebarLink link={links[5]}/>
        <SidebarLink link={links[6]}/>
        <SidebarLink link={links[7]}/>
      </nav>
    </div>
  )
};

Sidebar.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired
};

const mapStateToProps = (store) => ({
  user: store.user.data
});

// the { pure: false } parameter is used to fix the active link with react-redux
// https://github.com/ReactTraining/react-router/issues/3536
export default connect(mapStateToProps, null, null, { pure: false })(Sidebar);
