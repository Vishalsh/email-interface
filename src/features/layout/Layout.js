import React from 'react';
import { Route, Switch } from 'react-router-dom';

import routes from 'constants/routes';
import Header from 'features/header/Header';
import Sidebar from 'features/sidebar/Sidebar';
import Mailbox from 'pages/mailbox/Mailbox';

import classes from './Layout.module.scss';

const Layout = () => (
  <>
  <div className={classes.header}>
    <Header />
  </div>
  <div className={classes.sidebar}>
    <Sidebar />
  </div>

  <main className={classes.main}>
    <Switch>
      <Route exact path="/" component={Mailbox}/>
      <Route path={routes.DASHBOARD} component={() => <h1>DASHBOARD</h1>}/>
      <Route path={routes.LAYOUTS} component={() => <h1>LAYOUTS</h1>}/>
      <Route path={routes.GRAPHS} component={() => <h1>GRAPHS</h1>}/>
      <Route path={routes.MAILBOX} component={Mailbox}/>
      <Route path={routes.METRICS} component={() => <h1>METRICS</h1>}/>
      <Route path={routes.WIDGETS} component={() => <h1>WIDGETS</h1>}/>
      <Route path={routes.FORMS} component={() => <h1>FORMS</h1>}/>
      <Route path={routes.APP_VIEWS} component={() => <h1>APP VIEWS</h1>}/>
    </Switch>
  </main>
  </>
);

export default Layout;
