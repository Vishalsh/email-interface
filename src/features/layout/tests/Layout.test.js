import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Layout from '../Layout';
import Header from 'features/header/Header';
import Sidebar from 'features/sidebar/Sidebar';

describe('Layout', () => {
  const props = {
    children: <div className="routes"/>
  };
  const component = shallow(<Layout {...props}/>);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should toggle the sidebar', () => {
    component.find(Header).props().onClickHamburger();

    expect(component.state().isSidebarOpen).toBeTruthy();
    expect(component.find(Sidebar).props().isOpen).toBeTruthy();
  });
});