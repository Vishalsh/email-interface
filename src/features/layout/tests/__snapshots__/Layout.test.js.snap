// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the component 1`] = `
<div
  className="wrapper"
>
  <Connect(Sidebar)
    isOpen={false}
  />
  <div
    className="right"
  >
    <Header
      onClickHamburger={[Function]}
    />
    <main
      className="main"
    >
      <div
        className="routes"
      />
    </main>
  </div>
</div>
`;
